### PR TITLE
Generics

### PR DESCRIPTION
**Pull request by Manu0912**
# Summary
In this pull request i implemented generics to save in a container class a lis of Clients, Cars or Rentals. Also i created Client and Rental classes.

## Specifics
- 3 Classes have been created: Client.cs, Rental.cs and Container.cs
- Implement Generics in Container.cs to replace CarCRUD.cs 
- Delete CarCRUD.cs
- New routes in appsettings.json
- Implement the function GetAll()

### CODE REVIEW

File: AgenciaAutos/AgenciaAutos/appsettings.json [hunk 1]
```diff
@@ -1,17 +1,11 @@
...
     "Enabled": true,
     "AutoRetryDelay": "00:00:07"
   },
-  "Logging": {
-    "LogLevel": {
-      "Default": "Information",
-      "Microsoft": "Warning",
-      "Microsoft.Hosting.Lifetime": "Information"
-    }
-  },
```
Comment (jrocaspana) [inline AgenciaAutos/AgenciaAutos/appsettings.json]: I'm curious about why you removed the logging configuration but not the `TransientFaultHandlingOptions` section

File: AgenciaAutos/AgenciaAutos/Classes/Client.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+
+namespace AgenciaAutos
+{
+    class Client
```
Comment (jrocaspana) [inline AgenciaAutos/AgenciaAutos/Classes/Client.cs]: we will need to revise and follow conventions (applies to the entire class)
watch out for unnecessary and also missing empty lines!!

File: AgenciaAutos/AgenciaAutos/Classes/Container.cs [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+
+        public T Update(T item, string filePath)
+        {
+            T obj = default;
+            ReadJson(filePath);
+
+            List<Car> cars = new List<Car>();
+            List<Rental> rentals = new List<Rental>();
+            List<Client> clients = new List<Client>();
```
Comment (jrocaspana) [inline AgenciaAutos/AgenciaAutos/Classes/Container.cs]: in your words, what are the benefits of using generics by following this approach? I don't want to tell you what to do but rather help you figure it out, we can talk today in our meeting!

**Review by jrocaspana — COMMENTED** [review]
nice work Emanuel! I really appreciate your effort and dedication, but I think we need to make some adjustments, you missed a little bit the generics implementation but I'm convinced you will get it right and working!! 
please please don't forget following code conventions and being consistent with the use of `var` and/or explicit types when necessary.

File: AgenciaAutos/AgenciaAutos/appsettings.json (hunk 1 above)
Comment (Manu0912) [reply AgenciaAutos/AgenciaAutos/appsettings.json]: i just forgot, in the next commit i will get this out

File: AgenciaAutos/AgenciaAutos/Classes/Container.cs (hunk 3 above)
Comment (Manu0912) [reply AgenciaAutos/AgenciaAutos/Classes/Container.cs]: I know that is a bad answer but i learnt generics at the university and i just replied what they taught me there. Is a container class to save objects of different classes but it has limitations for that it cannot be possible add elements of a random class, for that i checked the types of the objects in the cointainer class. 
But i would like to show me which is the most convenient way in the meeting. 😄

File: AgenciaAutos/AgenciaAutos/Classes/Client.cs (hunk 2 above)
Comment (Manu0912) [reply AgenciaAutos/AgenciaAutos/Classes/Client.cs]: okey, i will check it!! thank you for the feedback.

**Pull request closed without merging**